api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java

```diff
@@ -109,18 +107,25 @@ private void collectErrorMessages(final ValidationException e,
     public EventType evolve(final EventType original, final EventTypeBase eventType) throws SchemaEvolutionException {
         checkEvolutionIncompatibilities(original, eventType);
         final var version = original.getSchema().getVersion();
+        final var typeChanged = !original.getSchema().getType().
+                equals(eventType.getSchema().getType());
+
+        if(typeChanged) {
+            return bumpVersion(original, eventType, MAJOR, new Version(version));
+        }
+
         switch (eventType.getSchema().getType()) {
```

Comment:
why do we need `switch` if it looks the same for any `type` ?

Reply:
The parameters are same but the methods are different. `evolveJsonSchema()` vs `evolveAvroSchema()`

Reply:
ups, missed that